iphone/Classes/TiUIApplicationShortcutsProxy.m

```diff
@@ -141,13 +141,13 @@ - (void)removeDynamicShortcut:(id)identifier
     return;
   }
 
-  NSMutableArray *shortcuts = (NSMutableArray *)[UIApplication sharedApplication].shortcutItems;
+  NSMutableArray<UIApplicationShortcutItem *> *shortcuts = (NSMutableArray *)[UIApplication sharedApplication].shortcutItems.mutableCopy;
 
-  for (UIApplicationShortcutItem *item in shortcuts) {
-    if ([item.type isEqualToString:key]) {
-      [shortcuts removeObject:item];
+  [shortcuts enumerateObjectsUsingBlock:^(UIApplicationShortcutItem * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
+    if ([obj.type isEqualToString:key]) {
+      [shortcuts removeObject:obj];
```

Comment:
It's not a good idea to mutate an array while enumerating it. Consider storing the indices that need to be removed in a `NSMutableIndexSet` and then remove them using `removeObjectsAtIndexes`. I know it was already wrong in the first place but since you are touching it here anyway let's make it right ;)

Also any particular reason you switched to enumerating using block instead of `for (... in ...)`. I don't see any benefit but instead it makes it more verbose.

Reply:
Damn, you're right with the mutation, I oversaw that! Regarding the `enumerateObjectsUsingBlock `: It's the faster and more ObjC'ish way of iterating. Some inspiring words can be found [here](https://stackoverflow.com/questions/4486622/when-to-use-enumerateobjectsusingblock-vs-for).